go/libraries/events/event_flush.go

```diff
@@ -0,0 +1,147 @@
+	// actual grpc server config
+
+	// host := dEnv.Config.GetStringOrDefault(env.MetricsHost, env.DefaultMetricsHost)
+	// portStr := dEnv.Config.GetStringOrDefault(env.MetricsPort, env.DefaultMetricsPort)
+	// insecureStr := dEnv.Config.GetStringOrDefault(env.MetricsInsecure, "false")
+
+	// port, err := strconv.ParseUint(*portStr, 10, 16)
+
+	// if err != nil {
+	// 	log.Println(color.YellowString("The config value of '%s' is '%s' which is not a valid port.", env.MetricsPort, *portStr))
+	// 	return emitter
+	// }
+
+	// insecure, err := strconv.ParseBool(*insecureStr)
+
+	// if err != nil {
+	// 	log.Println(color.YellowString("The config value of '%s' is '%s' which is not a valid true/false value", env.MetricsInsecure, *insecureStr))
+	// }
+
+	// hostAndPort := fmt.Sprintf("%s:%d", *host, port)
+	// conn, _ := dEnv.GrpcConnWithCreds(hostAndPort, insecure, nil)
+
+	// local grpc for dev
+	hostAndPort := fmt.Sprintf("%s:%d", "localhost", 50051)
+	conn, _ := dEnv.GrpcConnWithCreds(hostAndPort, true, nil)
+
+	return NewGrpcEmitter(conn)
+}
+
+// NewEventGrpcFlush creates a new EventGrpcFlush
+func NewEventGrpcFlush(fs filesys.Filesys, userHomeDir string, doltDir string, dEnv *env.DoltEnv) *EventGrpcFlush {
+
+	fbp := NewFileBackedProc(fs, userHomeDir, doltDir)
+
+	if exists := fbp.EventsDirExists(); !exists {
+		// should panic?
+		panic(ErrEventsDataDir)
+	}
+
+	return &EventGrpcFlush{em: getGRPCEmitter(dEnv), fbp: fbp}
+}
+
+func checkFilenameAndExt(fs filesys.Filesys, path string) (bool, error) {
+	filename := filepath.Base(path)
+	ext := filepath.Ext(filename)
+
+	if ext != evtDataExt {
+		return false, nil
+	}
+
+	data, err := fs.ReadFile(path)
+
+	if err != nil {
+		return false, err
+	}
+
+	md5FromFilename := filename[:len(filename)-len(ext)]
+	md5FromData := MD5StrUrl(data)
+
+	if md5FromFilename != md5FromData {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func (egf *EventGrpcFlush) sendToServer(ctx context.Context, fs filesys.Filesys, size int64, localPath string) error {
+	isFileValid, err := checkFilenameAndExt(fs, localPath)
+
+	if isFileValid && err == nil {
+		ctx, cnclFn := context.WithDeadline(context.Background(), time.Now().Add(time.Second+500*time.Millisecond))
+		defer cnclFn()
+
+		data, err := fs.ReadFile(localPath)
+		if err != nil {
+			// log error
+			return err
+		}
+
+		req := &eventsapi.LogEventsRequest{}
+
+		if err := proto.Unmarshal(data, req); err != nil {
+			// log error
+			return err
+		}
+
+		if err := egf.em.SendLogEventsRequest(ctx, req); err != nil {
+			// log error
+			return err
+		}
+
+		return nil
+	}
+
+	return err
+}
+
+// FlushEvents sends event logs to the events server
+func (egf *EventGrpcFlush) FlushEvents() error {
+	dp := NewDirectoryProcessor(egf.fbp.ed.fs, egf.fbp.GetEventsDirPath(), 4, time.Second, time.Minute, egf.sendToServer)
+	time.Sleep(2 * time.Second)
+	if err := dp.Close(); err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
Think I need to implement a default params struct similar to how we do in the events server? That way we can use the default params, but later we can easily supply new ones...